pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -435,12 +435,26 @@ func preStopHookInit(volumeMount []corev1.VolumeMount) container.Modification {
 
 func mongodbContainer(version string, volumeMounts []corev1.VolumeMount) container.Modification {
 	mongoDbCommand := []string{
-		"/bin/sh",
+		"/bin/bash",
```

Comment:
seems like just changing to bash may solve the issue as it seems that bash will forward the sigterm to its children: see option 2 here
https://pracucci.com/graceful-shutdown-of-kubernetes-pods.html#:~:text=When%20a%20pod%20should%20be,the%20running%20application%20and%20exit.

Reply:
I actually read that article and tried using just bash before but unfortunately that didn't work. I've researched this a bit but haven't been able to come to a conclusion on how bash actually handles signals. Here's somebody claiming bash will ignore any signals when a child process is executing: https://unix.stackexchange.com/a/146770. This article suggests one needs to use `exec`: https://itnext.io/containers-terminating-with-grace-d19e0ce34290 which unfortunately we can't because of our pre-stop hook.